pkg/addons/applier.go

```diff
@@ -282,6 +298,86 @@ func (a *Applier) waitForKubernetes(ctx context.Context) error {
 	}
 }
 
+func spinForInstallation(ctx context.Context, cli client.Client) error {
+	installSpin := spinner.Start()
+	installSpin.Infof("Waiting for additional components to be ready")
+
+	bgCtx, cancel := context.WithCancel(ctx)
+	ch := make(chan embeddedclusterv1beta1.InstallationStatus)
+	defer cancel()
+	spinMut := sync.Mutex{}
+	go func() {
+		spinMut.Lock()
+		defer spinMut.Unlock()
+		for {
+			select {
+			case <-bgCtx.Done():
+				return
+			case meta, ok := <-ch:
+				if !ok {
+					return // channel closed
+				}
+
+				// figure out what to log
+				if meta.State != embeddedclusterv1beta1.InstallationStatePendingChartCreation {
```

Comment:
Can we just `continue/break` on this if statement ? That would keep the happy path left padded. Also, can `meta` be renamed to something more meaningful on this context?

Reply:
yeah will do